From 49f5330f783579d2c8e6fbab2505060eebbc7cc0 Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Thu, 26 Dec 2013 12:14:19 +0200
Subject: [PATCH] Added wifi scanning disable function

This is important for boards that don't have wifi on it since sometimes the busses are floating that forbids proper boot (kernel hangs trying to scan the bus)

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 arch/arm/mach-mx6/board-mx6q_microsom.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-mx6/board-mx6q_microsom.c b/arch/arm/mach-mx6/board-mx6q_microsom.c
index f458d74..e8d84df 100644
--- a/arch/arm/mach-mx6/board-mx6q_microsom.c
+++ b/arch/arm/mach-mx6/board-mx6q_microsom.c
@@ -93,6 +93,7 @@
 
 static struct clk *sata_clk;
 static int caam_enabled;
+static int disable_wifi_detect = 0;
 
 extern char *gp_reg_id;
 extern char *soc_reg_id;
@@ -752,7 +753,8 @@ void __init mx6_usom_board_init(void)
 	imx6q_add_perfmon(1);
 	imx6q_add_perfmon(2);
 	/* WLan and BT stuff */
-	imx6q_usom_init_wifi_bt();
+	if (disable_wifi_detect != 1)
+		imx6q_usom_init_wifi_bt();
 }
 
 extern void __iomem *twd_base;
@@ -792,3 +794,11 @@ void __init mx6q_usom_reserve(void)
 
 }
 
+static int __init disable_wifi_detect_func(char *p)
+{
+	disable_wifi_detect = 1;
+	return 0;
+}
+
+early_param("disable_wifi_detect", disable_wifi_detect_func);
+
-- 
1.8.5.1

